src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -30,7 +31,18 @@ public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTyp
         {
             Source = EndpointParameterSource.Query;
             AssigningCode = $"httpContext.Request.Query[\"{parameter.Name}\"]";
-            IsOptional = parameter.Type is INamedTypeSymbol parameterType && parameterType.NullableAnnotation == NullableAnnotation.Annotated;
+            IsOptional = parameter.Type is INamedTypeSymbol
+            {
+                NullableAnnotation: NullableAnnotation.Annotated
+            };
+        }
+        else if (parameter.HasAttributeImplementingInterface(fromServiceMetadataInterfaceType))
+        {
+            Source = EndpointParameterSource.Service;
+            IsOptional = parameter.Type is INamedTypeSymbol { NullableAnnotation: NullableAnnotation.Annotated } || parameter.HasExplicitDefaultValue;
```

Comment:
Nit: Since you already made this change above.

```suggestion
            IsOptional = parameter.Type is INamedTypeSymbol
            {
                NullableAnnotation: NullableAnnotation.Annotated
            } || parameter.HasExplicitDefaultValue;
```

Also, shouldn't the optional check for query parameters be the same and include `|| parameter.HasExplicitDefaultValue`? Or are we not changing that yet since we would never pass in the default value unless it happens to be null?

That might technically a problem for value type services too, but I don't think even RDF bothers with assigning proper default values for service dependencies.

Reply:
> Also, shouldn't the optional check for query parameters be the same and include || parameter.HasExplicitDefaultValue?

Yes, `FromQuery` should also support default values. I plan on adding this in a follow-up PR.